src/timezones/interpret.jl

```diff
@@ -0,0 +1,193 @@
+# TimeZone concepts used to disambiguate context of DateTimes
+# abstract UTC <: TimeZone  # Defined in Base.Dates
+abstract Local <: TimeZone
+
+function transition_range(local_dt::DateTime, tz::VariableTimeZone, ::Type{Local})
+    transitions = tz.transitions
+
+    # Determine the earliest and latest possible UTC DateTime
+    # that this local DateTime could be.
+    # TODO: Alternatively we should only look at the range of offsets available within
+    # this TimeZone.
+    earliest = local_dt + MIN_OFFSET
+    latest = local_dt + MAX_OFFSET
+
+    # Determine the earliest transition the local DateTime could
+    # occur within.
+    start = searchsortedlast(
+        transitions, earliest,
+        by=el -> isa(el, Transition) ? el.utc_datetime : el,
+    )
+    start = max(start, 1)
```

Comment:
I'm clamping the `start` to be >= 1. In the case where `earliest` precedes all transitions we can get 0 as the start:
```julia
julia> searchsortedlast([2,3,4], 1)
0
```